distributed/tests/test_client.py

```diff
@@ -913,6 +913,31 @@ async def test_restrictions_ip_port(c, s, a, b):
     assert y.key in b.data
 
 
+@gen_cluster(client=True)
+async def test_restrictions_ip_port_task_key(c, s, a, b):
+    # Create a long dependency list
+    tasks = [delayed(inc)(1)]
+    for _ in range(100):
+        tasks.append(delayed(add)(tasks[-1], random.choice(tasks)))
+
+    last_task = tasks[-1]
+
+    # calculate all dependency keys
+    all_dependencies = list(last_task.dask.dependencies.keys())
```

Comment:
```suggestion
    all_dependencies = list(last_task.__dask_graph__())
```

This should be a more stable API to depend on.

Reply:
And maybe also rename `all_dependencies` to `all_tasks`?